source/SIL.AppBuilder.Portal.Frontend/src/ui/components/inputs/locale-input/-utils/helpers.tsx

```diff
@@ -0,0 +1,158 @@
+import * as React from 'react';
+
+export const hasSearchTerm = (searchTerm: string) => {
+  let lowerSearch = (searchTerm || '').trim().toLocaleLowerCase();
+
+  return (property: string) => {
+    try {
+      return (property || '').toLocaleLowerCase().includes(lowerSearch);
+    } catch (e) {
+      console.error(`searched for ${searchTerm} against value: ${property}`, e);
+    }
+  };
+};
+
+export const getSuggestions = (data: ILanguageInfo[]) => (value) => {
+  const has = hasSearchTerm(value);
+
+  return data.filter(
+    ({ name, nameInLocale, region, tag, localname, regions, names }) =>
+      has(nameInLocale) ||
+      has((names || []).join()) ||
+      has(localname || name) ||
+      has(regions || region) ||
+      has(tag)
+  );
+};
+
+export const getSuggestionValue = (suggestion: ILanguageInfo) => suggestion.tag;
+
+export function findLanguageCode(data: ILanguageInfo[]) {
+  return (value: string) => {
+    const info = data.find((lang) => lang.tag === value);
+
+    return (info && info.tag) || '';
+  };
+}
+
+export function findLanguageInfo(data: ILanguageInfo[]) {
+  return (value: string) => {
```

Comment:
as a one liner?
```
return (value: string) => data.find((lang) => lang.tag === value);
```

Reply:
yeah, it could be. I've been hesitant to go overboard with one liners, because it makes debugging annoying (gotta re-convert to a `{}` function, add `debugger` (or a breakpoint), then convert back, etc.
idk

Reply:
I hear ya. definitely not a deal breaker by any means